Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -48,28 +48,87 @@ namespace Hazel {
 			s_GLFWInitialized = true;
 		}
 
-		m_Window = glfwCreateWindow((int)props.Width, (int)props.Height, m_Data.Title.c_str(), nullptr, nullptr);
+		GLFWmonitor* windowRenderScreen = glfwGetPrimaryMonitor(); // where the window will be rendered
+		const GLFWvidmode* baseVideoMode = glfwGetVideoMode(windowRenderScreen);
+
+		// Set the actual width and height
+		switch (m_Data.Mode)
+		{
+			case WindowMode::Borderless:
+			case WindowMode::Fullscreen:
+				m_Data.Width = baseVideoMode->width;
+				m_Data.Height = baseVideoMode->height;
+				break;
+			case WindowMode::Windowed:
+				m_Data.Width = props.Width;
+				m_Data.Height = props.Height;
+				break;
+		}
+
+		HZ_CORE_INFO("Creating window '{0}' ({1}, {2})", m_Data.Title, m_Data.Width, m_Data.Height);
 
+		// Create window in windowed mode
+		glfwWindowHint(GLFW_DECORATED, m_Data.Mode == WindowMode::Windowed ? GLFW_TRUE : GLFW_FALSE);
+		glfwWindowHint(GLFW_RESIZABLE, m_Data.Mode == WindowMode::Windowed ? GLFW_TRUE : GLFW_FALSE);
+		glfwWindowHint(GLFW_FLOATING, m_Data.Mode == WindowMode::Fullscreen ? GLFW_TRUE : GLFW_FALSE);
+		glfwWindowHint(GLFW_AUTO_ICONIFY, m_Data.Mode == WindowMode::Fullscreen ? GLFW_TRUE : GLFW_FALSE);
+		m_Window = glfwCreateWindow((int)m_Data.Width, (int)m_Data.Height, m_Data.Title.c_str(),
+			m_Data.Mode == WindowMode::Windowed ? nullptr : windowRenderScreen, nullptr);
+
+		if (m_Data.Mode == WindowMode::Windowed)
+		{
+			// Extract the position in windowed mode
+			int Xpos, Ypos;
+			glfwGetWindowPos(m_Window, &Xpos, &Ypos);
+			m_Data.WindowedPos = { Xpos, Ypos };
```

Comment:
We should always initialize m_Data.WindowedPos, even if it is to 0...
We can initialize them directly, no need for local variables: 
`glfwGetWindowPos(m_Window, &m_Data.WindowedPos.x, &m_Data.WindowedPos.y);`